concepts/language-model.md

```diff
@@ -0,0 +1,58 @@
+---
+description: Language model
+---
+
+# Language model
+
+A language model takes text input (usually a history) and outputs the next word or character.
+
+Example:
+- input: `the man is riding a ___`
+- output: `bike` 40%, `car`: 30%, `scooter`: 10%, `blue`: 0.5%, ...
+
+In the example this would mean that the article `bike` has a 40% probability of being the continuation and is the most likely.
+
+It is possible to do *decoding*, which means repeatedly taking e.g. the most probable output, adding it to the text and so on.
+This way a language model can generate new text.
+
+# Models
+
+There are many different ways in which language models are created.
+
+## N-gram maximum likelihood estimate
+
+
+The easiest one is to count a number of occurrences of the pair 
```

Comment:
I think we should avoid this type of long-term dependency (the easiest what?  forces us to look across sections) and subjectivity (easiest), and be more factual, like an encyclopaedia.

> An **n-gram maximum likelihood estimate** is the count of occurrences of the phrase followed by that word, divided by the count of all occurrences of the phrase.

> > Formula
> > ...

(Phrase not pair, if we say n-gram not bi-gram, right?)

Reply:
I wanted to illustrate it on bigrams specifically but I can turn it into phrases.

Reply:
Yeah, keep it neutral, even if it happens to be a pair when n=2.

Reply:
Updated.